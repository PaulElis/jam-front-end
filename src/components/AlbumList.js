import React, { Component } from 'react';
import AlbumDetail from './AlbumDetail.js'
import '../styles/AlbumList.css'
// import MediaQuery from 'react-responsive';

import { getFavorites, deleteArtistFromFavorites, runSearch } from '../actions/actions'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

class AlbumList extends Component {
  state = {
    // albums: [],
    // favorites: [],
    // top_artists: [],
  }

  static getDerivedStateFromProps = (nextProps, prevState) => {
    return {
      albums: nextProps.albums === [] ? [] : nextProps.albums,
      // favorites: nextProps.favorites === [] ? [] : nextProps.favorites,
      top_artists: nextProps.top_artists === [] ? [] : nextProps.top_artists,
    }
  }

  componentDidMount = () => {

  }

  renderFavorites = () => {
    // console.log('in renderFavorites');
    return this.props.favorites.map(favorite =>
      <AlbumDetail key={favorite.url} artist={favorite} deleteArtist={this.props.deleteArtist} favorite={favorite} image={favorite.image}/>)
  }

  renderTopArtists = () => {
    // console.log('in renderTopArtists', this.state.top_artists);
    return this.state.top_artists.map(artist =>
      <AlbumDetail key={artist.url} artist={artist} image={artist.image[3]['#text']}/>
    )
  }

  renderAlbums = () => {
    return this.state.albums.map(album =>
      <AlbumDetail key={album.url} album={album} image={album.image[3]['#text']}/>)
  }

  render() {
    // console.log('AlbumList state:', this.state)
    // console.log('AlbumList props', this.props)
    return (
      <div id='albumlist-container'>
        {this.props.favorites ?
          this.renderFavorites() : this.state.top_artists ?
          this.renderTopArtists() : this.renderAlbums()}
      </div>
    );
  }
}

function mapStateToProps(state){
  return {
    albums: state.albums,
    // favorites: state.favorites,
    top_artists: state.top_artists,
  }
}

export default withRouter(connect(mapStateToProps, {getFavorites, deleteArtistFromFavorites, runSearch})(AlbumList))
