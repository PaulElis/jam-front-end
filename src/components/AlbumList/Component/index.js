import React, { Component } from 'react';
import AlbumDetailContainer from '../../AlbumDetail/Container'
import './index.css'

import { getFavorites, deleteArtistFromFavorites, runSearch } from '../../../actions/actions'
import { connect } from 'react-redux'

class AlbumList extends Component {

  render() {
    // console.log('AlbumList:', this.props);
    return (
      <div>
        <AlbumDetailContainer {...this.props} />
      </div>
    );
  }
}

function mapStateToProps(state){
  return {
    albums: state.albums,
    top_artists: state.top_artists,
  }
}

export default connect(mapStateToProps, {getFavorites, deleteArtistFromFavorites, runSearch})(AlbumList)
