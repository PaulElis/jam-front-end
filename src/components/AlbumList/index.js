import React, { Component } from 'react';
import AlbumDetail from '../AlbumDetail'
import './index.css'

import { getFavorites, deleteArtistFromFavorites, runSearch } from '../../actions/actions'
import { connect } from 'react-redux'

class AlbumList extends Component {

  render() {
    // console.log('AlbumList state:', this.state)
    // console.log('AlbumList props', this.props)
    // {this.state.top_artists ?
    //   this.renderTopArtists() : this.renderAlbums()}
    return (
      <div>
        <AlbumDetail {...this.props} />
      </div>
    );
  }
}

function mapStateToProps(state){
  return {
    albums: state.albums,
    top_artists: state.top_artists,
  }
}

export default connect(mapStateToProps, {getFavorites, deleteArtistFromFavorites, runSearch})(AlbumList)
