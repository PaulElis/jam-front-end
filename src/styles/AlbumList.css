div#albumlist-container{
  padding: 5%;
}
