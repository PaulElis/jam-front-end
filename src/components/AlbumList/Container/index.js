import React, { Component } from 'react';
import AlbumList from '../../AlbumList'
import './index.css'

import { getFavorites, deleteArtistFromFavorites, runSearch } from '../../../actions/actions'
import { connect } from 'react-redux'

class AlbumListContainer extends Component {

  state = {}

  static getDerivedStateFromProps = (nextProps, prevState) => {
    return {
      albums: nextProps.albums === [] ? [] : nextProps.albums,
      top_artists: nextProps.top_artists === [] ? [] : nextProps.top_artists,
    }
  }

  renderAlbumList = () => {
    console.log('in renderAlbumList', this.state.top_artists);
    if(this.state.top_artists){
      return this.state.top_artists.map(artist =>
      <AlbumList key={artist.url} artist={artist} top_artists={this.state.top_artists} image={artist.image[3]['#text']}/>
    )} else if (this.state.albums){
      return this.state.albums.map(album =>
      <AlbumList key={album.url} album={album} image={album.image[3]['#text']}/>
    )}
  }

  render() {
    return(
      <div id='albumlist-container'>
        {this.renderAlbumList()}
      </div>
    )
  }
}

function mapStateToProps(state){
  return {
    albums: state.albums,
    top_artists: state.top_artists,
  }
}

export default connect(mapStateToProps, {getFavorites, deleteArtistFromFavorites, runSearch})(AlbumListContainer)
