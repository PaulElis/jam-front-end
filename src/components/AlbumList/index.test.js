const index = require('./index');

import '../../setupEnzyme'
import { shallow } from 'enzyme';
import React from 'react';
import AlbumList from './index';

let wrapper

beforeEach(() => {
  const mockProps = {
    deleteArtistFromFavorites: jest.fn(),
    getFavorites: jest.fn(),
    runSearch: jest.fn(),
    history: {},
    location: {},
    match: {},
    albums: [],
    top_artists: [],
  }

  wrapper = shallow(<AlbumList {...mockProps} />)
})

describe('AlbumList', () => {
  it('snapshot AlbumList component', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
