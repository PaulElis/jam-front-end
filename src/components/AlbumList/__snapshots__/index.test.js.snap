// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList snapshot AlbumList component 1`] = `undefined`;
